File: loginapp-gui/src/app/models/product.ts
export interface Product {
  id: number
  productName: string
  serialNumber: string
  description: string
  itemCondition: ProductCondition
  addedTime: Date
}

export enum ProductCondition {
  NEW, GOOD, BAD, BROKEN
}

export interface ProductConditionSelect {
  name: string
  value: ProductCondition
}

export const ProductConditionList: ProductConditionSelect[] = [
  { name: "NEW", value: ProductCondition.NEW },
  { name: "GOOD", value: ProductCondition.GOOD },
  { name: "BAD", value: ProductCondition.BAD },
  { name: "BROKEN", value: ProductCondition.BROKEN }
]

export interface ProductForm {
  productName: string
  serialNumber: string
  description: string
  itemCondition: ProductConditionSelect
}

File: loginapp/src/main/java/edu/example/loginapp/products/entities/Product.java
package edu.example.loginapp.products.entities;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "products")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String productName;

    @Column(unique = true)
    private String serialNumber;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    private ItemCondition itemCondition;

    private LocalDateTime addedTime;

}
